lib/src/test/kotlin/net/dean/jraw/test/integration/SubmissionReferenceTest.kt

```diff
@@ -194,4 +194,33 @@ class SubmissionReferenceTest : Spek({
             SharedObjects.submittedSelfPost!!.sendReplies(true)
         }
     }
+
+    describe("set spoiler") {
+        assume({ SharedObjects.submittedSelfPost != null }, description = "should have a self post created") {}
+        it("should set/unset the post spoiler flair") {
+            SharedObjects.submittedSelfPost!!.setPostSpoiler(true)
+            SharedObjects.submittedSelfPost!!.inspect().isSpoiler.should.be.`true`
+            SharedObjects.submittedSelfPost!!.setPostSpoiler(false)
+            SharedObjects.submittedSelfPost!!.inspect().isSpoiler.should.be.`false`
+        }
+    }
+
+    describe("sticky post submission") {
+        assume({ SharedObjects.submittedSelfPost != null }, description = "should have a self post created") {}
+        it("should sticky a post into one of the allotted announcement slots") {
+            SharedObjects.submittedSelfPost!!.stickyPost(true)
+            SharedObjects.submittedSelfPost!!.inspect().isStickied.should.be.`true`
+        }
+
+        it("should fail trying to sticky a post to a nonexistent announcement slot") {
+            expectException(IllegalArgumentException::class) {
+                SharedObjects.submittedSelfPost!!.stickyPost(true, 4)
+            }
+        }
+
+        it("should remove the announcement status of the post") {
+            SharedObjects.submittedSelfPost!!.stickyPost(false)
+            SharedObjects.submittedSelfPost!!.inspect().isStickied.should.be.`false`
+        }
```

Comment:
The first and third tests can be combined into one `assume` block (see previous comments). The second test should be unnecessary

Reply:
What I should have said is that the *bodies* of the first and third tests can be combined. `assume` is basically a conditional `it` that skips the test if the given condition is false. I really wanted something like this:

```kt
assume({ SharedObjects.submittedSelfPost != null }, description = "should set/unset the post as a sticky") {
    val post = SharedObject.submittedSelfPost!!
    post.stickyPost(true)
    post.inspect().isStickied.should.be.`true`
    post.stickyPost(false)              
    post.inspect().isStickied.should.be.`false`
}
```

This is equivalent to writing

```kt
 if (SharedObjects.submittedSelfPost != null) {
    it("should set/unset the post as a sticky") {
        // test body
    }
} else {
    xit("should set/unset the post as a sticky", reason = "assumption failed") {
        // test body
    }
}
```